rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -77,23 +80,67 @@ def from_db(node: AgentNode):
         return obj
 
 
+class ExecutionMeta(BaseDbModel):
+    execution_id: str
+    started_at: datetime
+    ended_at: datetime
+    duration: float
+    total_run_time: float
+    status: autogpt_server.data.execution.ExecutionStatus
+
+    @staticmethod
+    def from_agent_graph_execution(execution: AgentGraphExecution):
+        now = datetime.now(timezone.utc)
+        start_time = execution.startedAt or execution.createdAt
+        end_time = execution.updatedAt or now
+        duration = (end_time - start_time).total_seconds()
+
+        total_run_time = 0
+        if execution.AgentNodeExecutions:
+            for node_execution in execution.AgentNodeExecutions:
+                node_start = node_execution.startedTime or now
+                node_end = node_execution.endedTime or now
+                total_run_time += (node_end - node_start).total_seconds()
+
+        return ExecutionMeta(
+            id=execution.id,
+            execution_id=execution.id,
+            started_at=start_time,
+            ended_at=end_time,
+            duration=duration,
+            total_run_time=total_run_time,
+            status=autogpt_server.data.execution.ExecutionStatus(
+                execution.executionStatus
+            ),
+        )
+
+
 class GraphMeta(BaseDbModel):
     version: int = 1
     is_active: bool = True
     is_template: bool = False
-
     name: str
     description: str
+    executions: list[ExecutionMeta] = []
```

Comment:
```suggestion
    executions: Optional[list[ExecutionMeta]] = None
```
To distinguish between a graph not having any executions vs the executions not being queried.